K2Bridge/KustoDAL/KustoDataAccess.cs

```diff
@@ -147,22 +152,96 @@ public async Task<ResolveIndexResponse> ResolveIndexAsync(string indexName)
             return response;
         }
 
-        private void MapFieldCaps(IDataReader kustoResults, FieldCapabilityResponse response)
+        private async Task MapFieldCaps(IDataReader kustoResults, FieldCapabilityResponse response, string tableName)
         {
             while (kustoResults.Read())
             {
                 var fieldCapabilityElement = FieldCapabilityElementFactory.CreateFromDataRecord(kustoResults);
-                if (string.IsNullOrEmpty(fieldCapabilityElement.Type))
+
+                if (fieldCapabilityElement.Type == "object")
+                {
+                    await HandleDynamicField(response, tableName, fieldCapabilityElement);
+                }
+                else
                 {
-                    Logger.LogWarning("Field: {@fieldCapabilityElement} doesn't have a type.", fieldCapabilityElement);
+                    if (string.IsNullOrEmpty(fieldCapabilityElement.Type))
+                    {
+                        Logger.LogWarning("Field: {@fieldCapabilityElement} doesn't have a type.", fieldCapabilityElement);
+                    }
+
+                    response.AddField(fieldCapabilityElement);
```

Comment:
So if no type was given, string will be chosen by default?

Reply:
This logic was here before this PR.
If it doesn't have a type it defaults to "unknown"